include/tvm/relay/expr.h

```diff
@@ -625,6 +629,64 @@ class TempExpr : public Expr {
   TVM_DEFINE_OBJECT_REF_METHODS(TempExpr, RelayExpr, TempExprNode);
 };
 
+/*
+ * Non-recursive traversal with dismantling unused call nodes,
+ * a derivative from ExpandDataflow method
+ */
+inline void __dismantle(Expr expr) {
+  std::unordered_set<const RelayExprNode*> visited;
+  std::stack<std::pair<Expr, bool>> stack;
+  auto fpush_to_stack = [&visited, &stack](const Expr& expr) {
+    if (expr.use_count() < 3 && !visited.count(expr.get())) {
+      stack.push({expr, false});
+    }
+  };
+  fpush_to_stack(expr);
+  while (stack.size() > 0) {
+    auto node = stack.top().first;
+
+    if (visited.count(node.get())) {
+      stack.pop();
+
+      // dismantle node
+      if (node.use_count() < 3) {
```

Comment:
could you elaborate a bit why number 3 and 2?

Reply:
Agreed. This needs a comment

Reply:
Commented in code